pointcloud/basic_octree_utils.go

```diff
@@ -136,12 +142,52 @@ func (octree *BasicOctree) helperSet(p r3.Vector, d Data, recursionDepth int) er
 		// Update metadata
 		octree.meta.Merge(p, d)
 		octree.size++
-		octree.node = newLeafNodeFilled(p, d)
+		octree.node = newLeafNodeFilled(p, d, octree.node.root)
+
+		// case where we add the first leaf to the octree
+		if recursionDepth == 0 {
+			return nil
+		}
+
+		nodeMPC := octree.node.maxChildProb
+		rootMPC := octree.node.root.node.maxChildProb
+		// only update tree if the new probability is greater than the node's root
+		if nodeMPC < rootMPC {
+			return nil
+		}
+		octDepth := getDepth(octree.node, 0)
+		err := updateChildProb(octree.node, octDepth)
+		return err
 	}
 
 	return nil
 }
 
+// updateChildProb recursively updates the max probability child of the node's root.
+func updateChildProb(node basicOctreeNode, octDepth int) error {
```

Comment:
[major] This feature may be more cleanly implemented as a part of helperSet, where you return an error and a max probability. Then in the case of an `internalNode` check the `err == nil` and update the max probability in that node (max(current max prob, new prob) and keep passing it back up and out of the octree so that each node in the setting path gets updated.

Reply:
i think this is a good suggestion, but I want to talk through the implications of this change before doing it.
Specifically, `octree` fulfills the `pointcloud` interface, so a change to the signature of `helperSet` also affects `Set` itself.

is a change to the interface within the scope of this ticket?
I do not have much context on how pointclouds are currently in use, so I am unaware of any unwanted ramifications.

right off the bat (within `pointcloud/pointcloud_file.go/ReadPCDToBasicOctree` , we would need to fix:
```
basicOct, ok := (cloud).(*BasicOctree)
```
because 
```
impossible type assertion: (cloud).(*BasicOctree)
	*BasicOctree does not implement PointCloud (wrong type for method Set)
		have Set(r3.Vector, Data) (int, error)
		want Set(r3.Vector, Data) error
```

looking for opinions 👀

Reply:
hmm, I see what you're saying. A potential caveat of this is that `helperSet` is called by `Set`, and that octrees currently fulfill the `pointcloud` interface. So, for that reason, we cannot return both an error and a max prob  - without violating the expected function signature.